### ISSUE TITLE
get_facts can't pickle...

### ISSUE THREAD

**Issue by LukeDRussell** [ansible-network/cisco_ios#60]
get_facts soft fails for certain hosts. I've not been able to figure out a pattern yet. 

The playbook run just kinda stops one any affected hosts, without actually failing. Like it's in an infinite loop or something.

```yaml

  - name: use ansible-network provider roles to gather facts
    include_role:
      name: "{{ ansible_network_provider }}"
      tasks_from: get_facts.yaml
    vars:
      subset: 
        - default
    when: ansible_network_os == 'ios'
```

```sh

TASK [ansible-network.cisco_ios : run command and parse output] **********************************************************************************************
Traceback (most recent call last):
  File "/home/4032956/.pyenv/versions/3.6.6/lib/python3.6/multiprocessing/queues.py", line 234, in _feed
    obj = _ForkingPickler.dumps(obj)
  File "/home/4032956/.pyenv/versions/3.6.6/lib/python3.6/multiprocessing/reduction.py", line 51, in dumps
    cls(buf, protocol).dump(obj)
TypeError: can't pickle StrictUndefined objects
```

**Comment by LukeDRussell** [comment]
OK, so narrowed down the problem to Cisco 2921 routers.
TLDR: Regex for model needs to include `[Cc]` not just `c`

Making this a lot harder to diagnose was that the parser wasn't explicitly failing when model returned null. When testing this with Python 2.7 it shows an infinite loop until it hits maximum stack size. 

Fix is to change regex.

It would also be nice if Ansible core failed with more specifics around which variable is undefined.
i.e. `"Model" is Null`.

**Comment by LukeDRussell** [comment]
#77 was merged.
